### PR TITLE
Redesigned help UI to accomodate long tooltips

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Added a Detached Border view that can be triggered by a Boolean slot
- Added redesigned Help UI to SectionedDetailPropertyView that can accommodate long help texts
image

### CODE REVIEW

File: src/foam/u2/borders/DetachedBorder.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.u2.borders',
+  name: 'DetachedBorder',
```
Comment (kgrgreer) [inline src/foam/u2/borders/DetachedBorder.js:9]: Why is it called 'Detached'?

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 2]
```diff
@@ -250,59 +236,72 @@ foam.CLASS({
...
-                        data: '/images/inline-error-icon.svg',
-                        displayHeight: '16px',
-                        displayWidth: '16px'
-                      })
-                        .style({
-                          'justify-content': 'flex-start',
-                          'margin': '0 8px 0 0'
+                    .style({ 'flex-grow': 1, 'max-width': '100%' })
+                    .callIf( prop.view.class == 'foam.u2.CheckBox', function() {
+                      this.addClass(this.myClass('CheckBoxWrapper'));
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionedDetailPropertyView.js:252]: Why is this needed?

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 3]
```diff
@@ -250,59 +236,72 @@ foam.CLASS({
...
+                        .show(errorSlot)
+                        .start({
+                          class: 'foam.u2.tag.Image',
+                          data: '/images/inline-error-icon.svg',
+                          displayHeight: '16px',
+                          displayWidth: '16px'
+                        })
+                          .style({
+                            'justify-content': 'flex-start',
+                            'margin': '0 8px 0 0'
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionedDetailPropertyView.js:287]: single quotes on margin not needed

File: src/foam/u2/borders/DetachedBorder.js (hunk 1 above)
Comment (sarthak-marwaha) [reply src/foam/u2/borders/DetachedBorder.js:9]: Because it is not linked to the trigger that controls its visibility.

File: src/foam/u2/detail/SectionedDetailPropertyView.js (hunk 2 above)
Comment (sarthak-marwaha) [reply src/foam/u2/detail/SectionedDetailPropertyView.js:252]: Used to align checkboxes and labels(in case they are present).

File: src/foam/u2/borders/DetachedBorder.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/u2/borders/DetachedBorder.js:9]: How is that evident from the code? Or is that just how it's used?

Comment (sarthak-marwaha) [reply src/foam/u2/borders/DetachedBorder.js:9]: Both really. In SectionedDetailPropertyView, the icon to show/hide it is not in the same div/parent unlike other collapsable borders. There is also no way to show/hide this view from within itself, which is something we do for other collapsing divs

Comment (kgrgreer) [reply src/foam/u2/borders/DetachedBorder.js:9]: In FOAM, "Detach" has a specific meaning, which makes this confusing. Can you think of a different name? Maybe ExpandableBorder?

File: src/foam/u2/detail/SectionedDetailPropertyView.js (hunk 2 above)
Comment (kgrgreer) [reply src/foam/u2/detail/SectionedDetailPropertyView.js:252]: Couldn't I just do the same with a better CSS selector instead of adding ChecBoxWrapper?

Comment (sarthak-marwaha) [reply src/foam/u2/detail/SectionedDetailPropertyView.js:252]: True, also can't think of any reason for this to live outside the CheckBox View.

File: src/foam/u2/borders/DetachedBorder.js (hunk 1 above)
Comment (sarthak-marwaha) [reply src/foam/u2/borders/DetachedBorder.js:9]: Works, will update
